pypeit/core/wavecal/fitting.py

```diff
@@ -194,7 +194,7 @@ def iterative_fitting(spec, tcent, ifit, IDs, llist, disp,
         # Keep unique ones
         ifit = np.unique(np.array(ifit, dtype=int))
         # Increment order
-        if n_order < (n_final+2):
+        if n_order < n_final:
```

Comment:
I don't think this is correct, as I mentioned to you before in Slack. This should be 
n_order <= n_final + 1. I do not want to deal with changing the polynomial order convention in all of our spectrograph files and archived fits, and this appears to be what you are doing here for a minor syntax change.

Reply:
This snippet was painfully confusing.  The intent is to run twice at the order=`n_final` specified by the user.  I don't believe the current implementation actually does this.  So I have fixed that and left the n_final+1.

Reply:
And 'fixed' yet again just now.

Reply:
I don't disagree with you that this snippet is terribly confusing. I did not write this piece of code. My concern is that you are effectively changing the n_final order of all of the pypeit archived solutions. You have not yet addressed that concern.

Reply:
Here is the wavelength QA for Shane Kast blue with the new algorithm.

![Arc_1dfit_A_1_01_S0000](https://user-images.githubusercontent.com/3998093/69554768-1fe69d00-0f57-11ea-8cc8-e91d2792a5a8.png)

Reply:
And X-Shooter, VIS 1x1, order 13

![Arc_1dfit_A_1_01_S0013](https://user-images.githubusercontent.com/3998093/69554879-4b698780-0f57-11ea-91db-60e63025fa03.png)

Reply:
So what is the concern that needs addressing?

Reply:
The concern is that you are effectively changing the maximum final order of all of the archived fits, basically lowering it by one. My suggestion is that you: 

1)  Either run all the tests, and visually inspect all the wavelength QA, i.e. all echelle orders for echelles. 

2) Or ensure that the final order of this new branch is the same as the final order of previous branches.